homepage rework and other improvements

juhp opened the pull request:
- rework the homepage layout and content
- enable navbar for homepage
- more navbar entries
- lts-16
- drop /?page
- hoogle-5.0.18
- improve /snapshots "buttons"
- improve snapshot top matter
- [q] better to use haskellstack.org for "get started"?


Opening this now for early review, comments, and feedback?
(I think it may need a little more hamlet/yesod foo, but I am reasonably happy so far.)

juhp commented:
![Screenshot from 2020-10-29 13-49-23](https://user-images.githubusercontent.com/82731/97530770-c8b23d80-19ed-11eb-8baf-6c376428a079.png)

juhp commented:
- (Btw I noticed for stackage.org, when I go above 200% zoom, the navbar entries go away)
- (Also stackage.org text rendering is pretty hideous on Chrome (I am using chromium) in dark mode)

juhp commented:
With normal default light mode desktop (120%)

![Screenshot from 2020-10-29 14-04-53](https://user-images.githubusercontent.com/82731/97531790-eaacbf80-19ef-11eb-86ff-52794ffdf63b.png)

snoyberg commented on src/Handler/Home.hs:
```diff
@@ -30,12 +30,8 @@ getHomeR :: Handler Html
 getHomeR = track "Handler.Snapshots.getAllSnapshotsR" $ do
     cacheSeconds $ 60 * 60
     now' <- getCurrentTime
-    currentPageMay <- lookupGetParam "page"
```
IIUC, this isn't a good idea. We're already having trouble with long database queries, isn't this disabling paging?

juhp replied:
I may be wrong, this is just for dropping the redundant https://www.stackage.org/?page=2 "feature".

snoyberg replied:
Doh, my mistake, sorry. Thanks for clarifying.

snoyberg commented on templates/default-layout.hamlet:
```diff
@@ -1,17 +1,15 @@
-        <span class="icon-bar">
-        <span class="icon-bar">
       <a class="brand" href="/">
-        <img src="/static/img/stackage.png" title="FP Complete">
```
Hah, good catch!

snoyberg commented on templates/default-layout.hamlet:
```diff
@@ -1,17 +1,15 @@
-          $forall route <- [AllSnapshotsR, BlogHomeR]
-            <li :Just route == cur:.active>
-              <a href=@{route}>#{route}
+            <li> <a href="/#about">about</a>
```
```suggestion
            <li> <a href="/#about">About</a>
```

snoyberg reviewed (commented):
Looking good! I wonder if the homepage is slightly too wordy with too much small-ish text. What do you think about collapsing to one column, increasing visibility of the important stuff, and leaving the more verbose "about" content to the end? Just an idea

juhp commented:
> Looking good! I wonder if the homepage is slightly too wordy with too much small-ish text.

I do agree - I was hesitating to delete too much text in one (or first) go. :-)
(So I tried "hiding" it at the bottom as a first attempt.)

> What do you think about collapsing to one column,

You mean not putting LTS and Snapshots side by side?

> increasing visibility of the important stuff, and leaving the more verbose "about" content to the end?

Yeah, we can try that.

juhp commented:
How about this now?:

Updated screenshot 133%
![Screenshot from 2020-10-30 21-37-20](https://user-images.githubusercontent.com/82731/97711524-292da180-1af8-11eb-92de-4eba785c89d0.png)

Still has 2 columns... but I add a summary at the top
(also removed links from the 2 section titles)
More details in the last commit.

juhp commented:
Seems, I broke the mobile navbar menu earlier...

juhp commented:
An alternative to Recent News could also be to just put the Blog post headline below the logo.

Another simplifying idea could be merging Recent News and Recent Snapshots (just listing the latest lts and nightly snapshot).

juhp commented:
Okay, @snoyberg, rows are gone now, and I combined few Recent Snapshots into News.

Apart from the mobile nav menu, I feel this is close to ready now.

With some placeholder snapshot texts to make it clearer:
![Screenshot from 2020-10-31 20-37-55](https://user-images.githubusercontent.com/82731/97779707-cfda7680-1bba-11eb-9959-ffd742223600.png)

I rather like this now.

juhp commented:
Okay I restored the navbar mobile menu too.

(I feel it kicks in too wide (early), but I think that is in the (modified) bootstrap library?)

snoyberg commented:
I think this is an improvement. My recommendation: I merge/release this, and then we iterate with others on further improvements as desired. What do you think?

Thanks for attacking this, I really like the changes!

juhp commented:
@snoyberg sounds good to me: I am all for further incremental improvements.

Might be good just to test it for a real build: though hopefully it works as expected.
